src/Commands/MeltorCommand.php

```diff
@@ -0,0 +1,309 @@
+<?php
+
+namespace Meltor\Commands;
+
+use Exception;
+use Illuminate\Console\Command;
+use Illuminate\Database\ConnectionInterface;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Meltor\Meltor;
+
+class MeltorCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'meltor:generate
+                {--f|force               : Execute without confirmation. }
+                {--i|ignoreProblems      : Leave out structures that cannot be processed. }
+                {--s|separateForeignKeys : Put foreign keys at the end of the file. Avoids problems with constraint checks. }
+                {--t|testrun             : Perform a db structure comparison testrun (will need to drop and restore your DB). }
+                {--r|restore             : Restore the db backup from a previous run, stops after that. }
+                ';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Create a new migration based on the current MySQL database';
+
+    // Internals
+    protected array $indexes = [];
+    protected array $problems = [];
+    protected array $spatialIndexes = [];
+    protected array $primaryKeyIsSetFor = [];
+    protected bool $canAccessInnoDbIndexes = false;
+    protected bool $ignoreProblems = false;
+    protected bool $warnAboutFloat = false;
+    protected bool $separateForeignKeys = false;
+    protected string $databaseName = '';
+    protected ?ConnectionInterface $dataConnection;
+    protected ?ConnectionInterface $schemaConnection;
+    protected ?Collection $structure;
+    protected ?Meltor $meltor;
+
+    /**
+     * Execute the console command.
+     *
+     * @return int
+     *
+     * @throws Exception
+     */
+    public function handle(): int
+    {
+        $this->meltor              = app('meltor');
```

Comment:
Weird not to use the facade

Reply:
this allows autocomplete in the IDE

Reply:
A better way would be to add phpdocs to your facade class. See the Laravel framework facades for reference.